### PR TITLE
Connect and Media API changes

### PR DESCRIPTION
**Pull request by manjeshbhargav**
@kwhinnery @markandrus ,

This PR contains the following changes:
* `Client` class has been removed and replaced with a `connect()` method.
* `getUserMedia()` has been replaced with `createLocalTracks()`.
* `createLocalAudioTrack()` method has been added to create a single audio track.
* `createLocalVideoTrack()` method has been added to create a single video track.
* `Media/LocalMedia` classes have been removed.
* `LocalParticipant` now maintains `tracks`, `audioTracks` and `videoTracks`.
* `LocalParticipant` has new methods `addTrack()` and `removeTrack()`.
* Fixed a bug where if you click on "preview my camera", then join a Room, leave, and join back, the preview wouldn't re-appear.

NOTE: These changes are being tested right now as part of `twilio-video#1.0.0-rc16`, and is scheduled to be pushed out with `twilio-video#1.0.0-beta5`. So you guys can review and suggest changes for now, and merge this PR once `beta5` is released.

### CODE REVIEW

File: public/quickstart.js [hunk 1]
```diff
@@ -1,9 +1,34 @@
...
+  }).forEach(function(mediaElements) {
+    mediaElements.forEach(function(mediaElement) {
+      mediaElement.remove();
+    });
+  });
+}
+
+function detachParticipantMedia(participant) {
+  detachMedia(Array.from(participant.tracks.values()));
+}
```
Comment (markandrus) [inline public/quickstart.js]: Maybe

* `attachTracks`
* `attachParticipantTracks` or `attachParticipant`
* `detachTracks`
* `detachParticipantTracks` or `detachParticipant`

?

File: public/quickstart.js [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+  attachMedia(Array.from(participant.tracks.values()), container);
+}
+
+function detachMedia(tracks) {
+  tracks.map(function(track) {
+    return track.detach();
+  }).forEach(function(mediaElements) {
+    mediaElements.forEach(function(mediaElement) {
+      mediaElement.remove();
+    });
```
Comment (markandrus) [inline public/quickstart.js]: Instead of two iterations of the loop in `attachMedia` and `detachMedia`, can we do a single?

**Comment by markandrus** [comment]
@manjeshbhargav two comments, otherwise +1

**Comment by jefflinwood** [comment]
Is this ok to merge @markandrus ?

**Comment by manjeshbhargav** [comment]
Hi @jefflinwood ,

The API changes will be available to customers as part of the `1.0.0-beta5` release. So I think, and @markandrus agrees, that we should hold off on merging this until then.

**Comment by jefflinwood** [comment]
Got it thanks @manjeshbhargav - I see that 1.0.0-beta4 is still the current version.

**Pull request merged**